src/tree.rs

```diff
@@ -579,14 +579,15 @@ where
         if let Some((child_nibble, child)) = it.next() {
             if it.next().is_none() && child.is_leaf() {
                 // internal node has only one child left and it's leaf node, replace it with the leaf node
-                let child_key = node_key.gen_child_node_key(child.version, *child_nibble);
+                let child_key = node_key.gen_child_node_key(child.version, child_nibble);
                 let child_node = tree_cache.get_node(&child_key)?;
                 tree_cache.delete_node(&child_key, true /* is_leaf */);
 
                 node_key.set_version(version);
                 tree_cache.put_node(node_key.clone(), child_node.clone())?;
                 Ok(PutResult::Updated((node_key, child_node)))
             } else {
+                drop(it);
```

Comment:
This is because the type of `iter()` changed from a concrete `HashMap::Iter` to `impl Iterator`. While the hashmap iterator doesn't implement drop, `dropck` can't assume that about `impl Iterator`.